pandas/core/generic.py

```diff
@@ -742,11 +742,13 @@ def swaplevel(self, i=-2, j=-1, axis=0):
 
         Parameters
         ----------
+        %(optional_mapper)s
         %(axes)s : scalar, list-like, dict-like or function, optional
             Scalar or list-like will alter the ``Series.name`` attribute,
```

Comment:
Is this also true for the mapper?

Reply:
This is the `Series.rename` docstring right now:

```
Signature: df.A.rename(index=None, **kwargs)
Docstring:
Alter axes input function or functions. Function / dict values must be
unique (1-to-1). Labels not contained in a dict / Series will be left
as-is. Extra labels listed don't throw an error. Alternatively, change
``Series.name`` with a scalar value (Series only).

Parameters
----------

index : scalar, list-like, dict-like or function, optional
    Scalar or list-like will alter the ``Series.name`` attribute,
    and raise on DataFrame or Panel.
    dict-like or functions are transformations to apply to
    that axis' values
```

So i didn't modify the signature at all. It doesn't have an `mapper` argument.

Reply:
(I think that not having `mapper` or `axis` kwargs is the better option here, but I may be wrong)

Reply:
Sorry, I wanted to say that for the DataFrame docstring (this is in generic.py). 
So I meant, for a dataframe.rename, is the explanation of index/column, also valid for mapper (scalar vs mapping)

Reply:
Yeah, the dict / function-like arguments descriptions are correct. Scalar / list-like don't apply. I could make that bit a shared_doc section... Let me try quick.

Reply:
Although, at some point a separate docstring might be easier.

Reply:
Ugh, this is pretty messy. I'm just going to leave as-is for now. For `DataFrame.rename` the docstring starts

```
Parameters
----------
mapper : dict-like or function
    Applied to the axis specified by `axis`
index, columns : scalar, list-like, dict-like or function, optional
    Scalar or list-like will alter the ``Series.name`` attribute,
    and raise on DataFrame or Panel.
    dict-like or functions are transformations to apply to
    that axis' values
axis : int or str, optional
    Axis to target. Can be either the axis name ('index', 'columns')
    or number (0, 1).
```

Reply:
I personally would also create separate docstrings for the series/dataframe versions of those methods. It will give a bit of duplication, but it will be much easier and take less time to make sure this duplication is done correctly than trying to figure out how to make it work with the docstring substitution. 
But agree that can be in another PR.